File: dev-superior-algoritmo-e-logica-de-programacao/java/src/exercicios/DiagonalNegativos.java
package exercicios;

import java.util.Locale;
import java.util.Scanner;

public class DiagonalNegativos {

	public static void main(String[] args) {

		Locale.setDefault(Locale.US);
		Scanner sc = new Scanner(System.in);

		int N, cont;

	    System.out.print("Qual a ordem da matriz? ");
		N = sc.nextInt();

	    int[][] mat = new int[N][N];

		for (int i = 0; i < N; i++) {
			for (int j = 0; j < N; j++) {
	        	System.out.printf("Elemento [%d,%d]: ", i, j);
				mat[i][j] = sc.nextInt();
	        }
	    }

		System.out.println("\nMATRIZ DIGITADA:");
		for (int i = 0; i < N; i++) {
			for (int j = 0; j < N; j++) {
				System.out.print(mat[i][j] + " ");
			}
			System.out.println();
		}

	    System.out.println("\nDIAGONAL PRINCIPAL:");
		for (int i = 0; i < N; i++) {
	    	System.out.printf("%d ", mat[i][i]);
	    }

		cont = 0;
		for (int i = 0; i < N; i++) {
			for (int j = 0; j < N; j++) {
				if (mat[i][j] < 0) {
					cont++;
				}
			}
		}

		if (cont == 0) {
			System.out.print("\n\nNao ha negativos na matriz");
		} else {
			System.out.printf("\n\nQUANTIDADE DE NEGATIVOS = %d\n", cont);
		}

		sc.close();
	}
}

File: nelio-alves-algoritmo-e-logica-de-programacao/python/exercicios/diagonal_negativos.py
N: int; cont: int

N = int(input("Qual a ordem da matriz? "))

mat: [[int]] = [[0 for x in range(N)] for x in range(N)]

for i in range(N):
	for j in range(N):
		mat[i][j] = int(input(f"Elemento [{i},{j}]: "))

print("\nMATRIZ DIGITADA:")
for i in range(N):
    for j in range(N):
        print(f"{mat[i][j]} ", end="")
    print()

print("\nDIAGONAL PRINCIPAL:")
for i in range(N):
	print(f"{mat[i][i]} ", end="")

cont = 0
for i in range(N):
	for j in range(N):
		if mat[i][j] < 0:
			cont = cont + 1

if cont == 0:
	print("\n\nNao ha negativos na matriz")
else:
	print(f"\n\nQUANTIDADE DE NEGATIVOS = {cont}")